install_dependencies.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2024 Tenstorrent, Inc. All rights reserved.
+#
+# This script is based on `xrtdeps.sh` from the Xilinx XRT project.
+# Original source: https://github.com/Xilinx/XRT/blob/master/src/runtime_src/tools/scripts/xrtdeps.sh
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FLAVOR=`grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"'`
```

Comment:
I like the concept of "just run a script".  My concern is dead code (and with it, miscommunication and future maintenance).

We don't support any other distros (yet).  But now it's dead code and suggests that this might work on other distros, which could lead people down a sad path.
We don't seem to be doing anything with `--docker`, but it's presented in the help message.

If we can KISS this script, then I'd be happy to see it used.

Reply:
I am okay to remove the docker flag until we need it.

I don't want to take out the distro checking.

What if someone set up there host with fedora? This way the script checks.

Plus we have a framework ready for future expansion.

Reply:
Sounds good.  yeah, I see later on that we'll `exit 1` for that.  I like that.